esileapclient/tests/unit/osc/v1/base.py

```diff
@@ -0,0 +1,32 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import mock
+
+from osc_lib.tests import utils
+
+
+class TestESILeapCommand(utils.TestCommand):
+
+    def setUp(self):
+        super(TestESILeapCommand, self).setUp()
+
+        self.app.client_manager.auth_ref = mock.Mock(auth_token="TOKEN")
+        self.app.client_manager.lease = mock.Mock()
+
+
+class FakeResource(object):
```

Comment:
I thought every resource needed a manager?

Reply:
In general this change to the tests is a bit tricky to understand

Reply:
Ah, so: `fakes.py` contained both test data and base classes used by all tests. I separated the two out, so that the owner change tests extend base classes and do not import `fakes.py`. 

Resources don't need managers; clients do. `lease` here is mocking the leasing client; you'll see this if you look at, say, the offers tests and drill down as to how `self.app.client_manager.lease` is used. I changed the variable names there from `lease_mock` to `client_mock` to make it more clear.

Let me know if this makes sense!

Reply:
Ok. Part of my confusion is that the fakes.TestLease referred to Lease in the old sense, that this is the leasing service. And the other TestLease was probably originally TestContract